### PR TITLE
async ticket handling

### PR DESCRIPTION
**Pull request by ktdreyer**
A couple of changes here, but the main features:
* use treq to avoid blocking the reactor
* match multiple tickets at once

### CODE REVIEW

**Comment by alfredodeza** [comment]
this PR is too big

**Comment by ktdreyer** [comment]
No problem, I can break it up into parts. Here's part 1: https://github.com/alfredodeza/helga-redmine/pull/11

**Comment by alfredodeza** [comment]
You should go on vacation again

**Comment by ktdreyer** [comment]
My favorite politician

"We Have to Merge the PR So That You Can Find Out What Is In It"

File: redmine.py [hunk 1]
```diff
@@ -1,74 +1,106 @@
...
-
-    try:
-        return result['issue']['subject']
-    except KeyError:
-        return 'unable to read subject'
+def match_tickets(message):
+    tickets = []
+    pattern = re.compile(r"""
+       (?:               # Prefix to trigger the plugin:
+            issues?      #   "issue" or "issues"
```
Comment (alfredodeza) [inline redmine.py]: wouldn't this match: iss, issu, is as well?

File: tests/test_redmine.py [hunk 2]
```diff
@@ -41,40 +43,91 @@ def fail_line_matrix():
     return lines
 
 
+multiple_ticket_lines = [
+    'issues #1 #2 #3',
```
Comment (alfredodeza) [inline tests/test_redmine.py]: this is not testing `issue`, `ticket`, `tickets`, `redmine`, `redmines`, `bug` and `bugs` (maybe it wasn't before but it is trivial to add them)

File: redmine.py (hunk 1 above)
Comment (ktdreyer) [inline redmine.py]: The `?` just makes the preceding token optional. So it's saying "make `s` optional" there.

File: tests/test_redmine.py (hunk 2 above)
Comment (ktdreyer) [inline tests/test_redmine.py]: Yeah, this test is not as nice as your matrix ones - it was over my head to set up the full matrix like you've got above, there.

**Comment by ktdreyer** [comment]
Sorry I missed that treq bumps the minimum Twisted version requirement. https://github.com/alfredodeza/helga-redmine/pull/12 fixes that.

**Comment by alfredodeza** [comment]
terrible

**Pull request merged**